### PR TITLE
Address inconsistencies when using git (packaged with gitaly)

### PR DESCRIPTION
**Pull request by sachilles**
Inconsistencies (see #2549) are addressed in this PR.

### CODE REVIEW

**Comment by kkimurak** [comment]
same as #2553 ?

**Comment by sachilles** [comment]
Yes, seems to be somehow parallel work (with the same result). ^^

**Comment by kkimurak** [comment]
All tests passed. thanks for work. 
By the way I have some comment but which one will be merged?

**Comment by sachilles** [comment]
Since I did remove one additional line (compared to #2553), feel free to commit to this PR. I guess @paolomainardi will understand and forgive me.

**Comment by kkimurak** [comment]
I see.

**Comment by sachilles** [comment]
@kkimurak I would like to test the image before merging. I hope that this will not pose any problems for us, even in the context of the recently published security release (see <https://about.gitlab.com/releases/2022/05/02/security-release-gitlab-14-10-1-released/>).

File: Dockerfile [hunk 1]
```diff
@@ -49,7 +49,7 @@ RUN set -ex && \
       libpq5 zlib1g libyaml-0-2 libssl1.1 \
       libgdbm6 libreadline8 libncurses5 libffi7 \
       libxml2 libxslt1.1 libcurl4 libicu66 libre2-dev tzdata unzip libimage-exiftool-perl \
-      libcurl4-openssl-dev libexpat1-dev gettext libz-dev libssl-dev libpcre2-dev build-essential git-core \
+      libcurl4-openssl-dev libexpat1-dev gettext libz-dev libssl-dev libpcre2-dev build-essential \
```
Comment (kkimurak) [inline Dockerfile:52]: These packages (added in #2553) can be moved to BUILD_DEPENDENCIES as well. This can reduce the size of the image by 200MB.

Comment (sachilles) [reply Dockerfile]: Thanks for this hint. Do you have any other comments or hints?

**Review by kkimurak — APPROVED** [review]
`build-essential` contains` gcc`, `g ++`, `libc6-dev`, and` make`, so I think you can remove one of them.
Anyway LGTM.
 
I built it locally and checked it, but I couldn't reproduce the problem reported #2549, so I'm not sure if there are any other major problems.

**Comment by sachilles** [comment]
@kkimurak What do you mean exactly by

> `build-essential` contains` gcc`, `g ++`, `libc6-dev`, and` make`, so I think you can remove one of them.
> Anyway LGTM.

**Comment by kkimurak** [comment]
`gcc g++ make libc6-dev` can be replaced by `build-essential` for simplicity but this is out of scope and may not caeses any issues. Sorry for not clear.

**Pull request merged**